search/query_language_whitebox_test.go

```diff
@@ -270,7 +271,76 @@ func TestParseMap(t *testing.T) {
 		assert.Equal(t, expectedQuery, actualQuery)
 	})
 
+	t.Run(OPTS, func(t *testing.T) {
+		t.Parallel()
+		// given
+		input := fmt.Sprintf(`{"%s": [ {"parent-exists": true}, {"tree-view": true}]}`, OPTS)
+		// Parsing/Unmarshalling JSON encoding/json
+		fm := map[string]interface{}{}
+		err := json.Unmarshal([]byte(input), &fm)
+		require.Nil(t, err)
+		// when
+		actualQuery := Query{}
+		parseOptions(fm, &actualQuery)
+		// then
+		qo := QueryOptions{ParentExists: true, TreeView: true}
+		expectedQuery := Query{Options: &qo}
+		assert.Equal(t, expectedQuery, actualQuery)
+	})
+	t.Run(OPTS+" complex query", func(t *testing.T) {
+		t.Parallel()
+		// given
+		input := fmt.Sprintf(`{"$AND":[{"title":"some"},{"state":"new"}],"%s": [ {"parent-exists": true}, {"tree-view": true}]}`, OPTS)
+		// Parsing/Unmarshalling JSON encoding/json
+		fm := map[string]interface{}{}
+		err := json.Unmarshal([]byte(input), &fm)
+		require.Nil(t, err)
+		// when
+		actualQuery := Query{}
+		parseOptions(fm, &actualQuery)
+		// then
+		qo := QueryOptions{ParentExists: true, TreeView: true}
+		expectedQuery := Query{Options: &qo}
+		assert.Equal(t, expectedQuery, actualQuery)
+
+		parseMap(fm, &actualQuery)
+		title := "some"
+		state := "new"
+		expectedQuery = Query{Options: &qo,
+			Name: AND, Children: []Query{
+				{Name: "title", Value: &title},
```

Comment:
This has nothing to do with your code but he many times we're creating variables just to get their pointers makes me think of a package `strptr` that just implements something like this:

```go
package strptr

// StringPtr returns the pointer of the given string s
func StringPtr(s string) *string {
    return &s
}
```